okhttp/src/main/java/okhttp3/Request.kt

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (C) 2013 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3
+
+import okhttp3.internal.Util
+import okhttp3.internal.http.HttpMethod
+import java.net.URL
+
+/**
+ * An HTTP request. Instances of this class are immutable if their [body] is `null` or itself
+ * immutable.
+ */
+class Request internal constructor(
+  internal val url: HttpUrl,
+  builder: Builder
+) {
+  internal val method: String = builder.method
+  internal val headers: Headers = builder.headers.build()
+  internal val body: RequestBody? = builder.body
+  internal val tags: Map<Class<*>, Any> = Util.immutableMap(builder.tags)
+
+  private val cacheControl: CacheControl by lazy {
+    CacheControl.parse(headers)
+  }
+
+  val isHttps: Boolean
+    get() = url.isHttps
+
+  fun url(): HttpUrl = url
+
+  fun method(): String = method
+
+  fun headers(): Headers = headers
+
+  fun header(name: String): String? = headers[name]
+
+  fun headers(name: String): List<String> = headers.values(name)
+
+  fun body(): RequestBody? = body
+
+  /**
+   * Returns the tag attached with `Object.class` as a key, or `null` if no tag is attached with
```

Comment:
Why are you putting null in backticks? Don’t do that.

We want to avoid backticks. Null is a normal word like tomato or taco or true, and it’s completely unambiguous without backticks.

Reply:
We always put `{@code null}` in our codebase since it's kind of like a constant value ref, so just was doing it out of habit. Will fix

Reply:
d853c19